### ISSUE TITLE
loop extension causes RTE when leaving view

### ISSUE THREAD

**Issue by aneeskA** [gskinner/flutter_animate#50]
I am using

```dart
.animate(
  onPlay: (controller) =>
   controller.loop(count: 2, reverse: true))
  .shimmer(
    delay: 400.ms,
    duration: 1800.ms,
    color: Colors.deepPurple.shade900)
  .shake(hz: 5, curve: Curves.easeInOutCubic)
  .scaleXY(end: 1.2, duration: 600.ms)
  .then(delay: 600.ms)
  .scaleXY(end: 1 / 1.1)
```

to animate an icon. When I navigate away from the page, the following crash is seen. How do I say cancel the pending animation?

```console
2023-01-29 19:39:42.737004+0530 Runner[15014:4806242] [VERBOSE-2:dart_vm_initializer.cc(41)] Unhandled Exception: 'package:flutter/src/animation/animation_controller.dart': Failed assertion: line 533 pos 7: '_ticker != null': AnimationController.animateTo() called after AnimationController.dispose()
AnimationController methods should not be used after calling dispose.
#0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
#2      AnimationController.animateTo (package:flutter/src/animation/animation_controller.dart:533:7)
#3      AnimationControllerLoopExtensions.loop.<anonymous closure> (package:flutter_animate/src/extensions/animation_controller_loop_extensions.dart:51:9)
#4      _RootZone.run (dart:async/zone.dart:1654:54)
#5      Future.timeout.<anonymous closure> (dart:async/future_impl.dart:865:34)
#6      Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:18:15)
#7      _Timer._runTimers (dart:isolate-patch/timer_impl.dart:398:19)
#8      _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
#9      _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:192:26)
2023-01-29 19:40:01.785378+0530 Runner[15014:4813608] [tcp] tcp_input [C12.1.1.10:3] flags=[R] seq=3463997066, ack=0, win=0 state=FIN_WAIT_1 rcv_nxt=3463997066, snd_una=4155211570
```

**Comment by aneeskA** [comment]
@gskinner I found the code from your recent video : https://www.youtube.com/watch?v=lxkFL9disq0&t=865s

**Comment by gskinner** [comment]
Thanks for the report, it looks like this is caused by the `AnimationController.loop` extension.

@Thithip — any chance you want to take a look at this, since it was your PR? If not, I'll take a stab at it later today / tomorrow.

**Comment by gskinner** [comment]
Just pushed a fix for this, it should also prevent `loop` from restarting in some cases after you call `AnimationController.stop()` or similar.

**Comment by Thithip** [comment]
> Thanks for the report, it looks like this is caused by the `AnimationController.loop` extension.
> 
> @Thithip — any chance you want to take a look at this, since it was your PR? If not, I'll take a stab at it later today / tomorrow.


You're too fast for me, it's already fixed!

Thank you @gskinner for this 👍 Next time I'll wake up sooner 😅

**Comment by aneeskA** [comment]
That was fast! Thank you so much for doing this. I tried out the change and I can confirm that the fix is working for me. 💪 

I guess now I wait for the official release to happen? I new to flutter ecosystem, thus the question.

**Comment by S-ecki** [comment]
@aneeskA  You could also choose to depend on this GitHub repo directly instead of the package. (And change it later if the fix is published).

See https://dart.dev/tools/pub/dependencies#git-packages for an explanation how.

**Comment by gskinner** [comment]
Correct. It will be included in the next update. I don't have a specific date for that, but I might kick out a quick dot release soon, before tackling some more bugs / features.

**Comment by gskinner** [comment]
Just special for you I just published v3.1, with this fix and a couple of other minor things. :)
